aemedge/blocks/quote/quote.js

```diff
@@ -1,12 +1,39 @@
-export default function decorate(block) {
+import { getAuthorEntries, removeAuthorsSuffixes } from '../../scripts/article.js';
+import Avatar from '../../libs/avatar/avatar.js';
+import { getMetadata } from '../../scripts/aem.js';
+
+export default async function decorate(block) {
   const qtContent = block.querySelector(':scope > div:first-of-type > div');
   const qsContent = block.querySelector(':scope > div:nth-of-type(2) > div');
+  const linkContent = block.querySelector(':scope > div:nth-of-type(3) > div');
+  const isNotArticle = getMetadata('template') !== 'article';
+  const isSmall = block.classList.contains('small');
   qtContent.classList.add('col', 'content');
   qtContent.parentNode.classList.add('qt');
   if (qsContent) {
     qsContent.classList.add('col', 'content');
     qsContent.parentNode.classList.add('qs');
-  } else {
+    if (isNotArticle) {
+      const author = removeAuthorsSuffixes(qsContent.textContent).split(',')[0].trim();
+      const authorEntry = (await getAuthorEntries([author]))[0];
+      if (authorEntry) {
+        const avatar = new Avatar(authorEntry.title, null, authorEntry.path, authorEntry.image);
+        block.insertBefore(avatar.render(isSmall ? 'medium' : 'big', false, true), qtContent.parentNode);
+      }
+    }
+  }
+  if (linkContent && isNotArticle) {
+    // Unwrap link from button container added by aem.js
+    const link = linkContent.querySelector('a');
+    link.classList = [];
```

Comment:
I think you could populate the list directly with the values from the next line, no?